recipes/libsodium/all/conanfile.py

```diff
@@ -118,7 +108,13 @@ def source(self):
 
     @property
     def _msvc_sln_folder(self):
-        if self.settings.compiler == "Visual Studio":
```

Comment:
I guess `str(self.settings.compiler) == "Visual Studio"` would also work?

Reply:
probably so

Reply:
there other checks like that afterwards, so it requires more modifications in this method.

Reply:
I would replace by something like this:

```python
    @property
    def _msvc_sln_folder(self):
        sln_folders = {
            "Visual Studio": {
                "10": "vs2010",
                "11": "vs2012",
                "12": "vs2013",
                "14": "vs2015",
                "15": "vs2017",
                "16": "vs2019",
            },
            "msvc": {
                "170": "vs2012",
                "180": "vs2013",
                "190": "vs2015",
                "191": "vs2017",
                "192": "vs2019",
            }
        }

        if self.version != "1.0.18":
            sln_folders["Visual Studio"]["17"] = "vs2022"
            sln_folders["msvc"]["193"] = "vs2022"

        sln_folder = sln_folders.get(str(self.settings.compiler), {}).get(str(self.settings.compiler.version))
        if not sln_folder:
            sln_folder = "vs2022" if self.version != "1.0.18" else "vs2019"

        return sln_folder
```

Reply:
Absolutely right, I did this quickly and didn't check it further. Thank you for the code suggestion